functions/src/on-update-contract.ts

```diff
@@ -55,6 +57,20 @@ async function handleResolvedContract(contract: Contract) {
     firestore.collection(`contracts/${contract.id}/comments`)
   )
 
+  // update user metrics for this contract for the final time
+  const betsByUser = groupBy(bets, (b) => b.userId)
+  await Promise.all(
+    Object.entries(betsByUser).map(([userId, userBets]) => {
+      const metrics = getContractBetMetrics(contract, userBets)
+      return firestore
+        .collection('users')
+        .doc(userId)
+        .collection('contract-metrics')
+        .doc(contract.id)
+        .set(metrics)
+    })
```

Comment:
So, `getContractBetMetrics` doesn't include all the fields, there's `from` which has the daily weekly monthly profit changes.

If we're updating markets for up to 30 days after resolution then I guess we could remove this code anyway.

Reply:
That's a good point. I think I will just remove it. That would have been simpler in the first place.